### PR TITLE
Mark a floating Task as Finished

### PR DESCRIPTION
**Pull request by nightism**
This PR implement the Command which can mark a floating task as finished.

### CODE REVIEW

File: src/main/java/seedu/address/model/task/DeadlineTask.java [hunk 1]
```diff
@@ -41,7 +41,8 @@ public boolean equals(Object other) {
         return other == this
                || (other instanceof DeadlineTask
                && name.equals(((DeadlineTask)other).name)
-               && due.equals(((DeadlineTask)other).due));
+               && due.equals(((DeadlineTask)other).due)
+               && finished == ((DeadlineTask)other).finished);
```
Comment (pyokagan) [inline src/main/java/seedu/address/model/task/DeadlineTask.java:45]: Hmph, I think the tests need to be changed to be more comprehensive.

File: src/test/java/seedu/address/model/task/DeadlineTaskTest.java [hunk 2]
```diff
@@ -43,9 +43,12 @@ public void equals_isEqual_returnsTrue() throws Exception {
...
     public void equals_notEqual_returnsFalse() throws Exception {
-        final DeadlineTask other = new DeadlineTask(new Name("Deadline Task Name"), UNIX_EPOCH.plusDays(2));
-        assertFalse(deadlineTask.equals(other));
-        assertFalse(deadlineTask.hashCode() == other.hashCode());
+        final DeadlineTask other1 = new DeadlineTask(new Name("Deadline Task Name"), UNIX_EPOCH.plusDays(2), false);
+        final DeadlineTask other2 = new DeadlineTask(new Name("Deadline Task Name"), UNIX_EPOCH.plusDays(1), true);
+        assertFalse(deadlineTask.equals(other1));
+        assertFalse(deadlineTask.hashCode() == other1.hashCode());
+        assertFalse(deadlineTask.equals(other2));
+        assertFalse(deadlineTask.hashCode() == other2.hashCode());
```
Comment (nightism) [inline src/test/java/seedu/address/model/task/DeadlineTaskTest.java:51]: @pyokagan is this what you expect? Or something else?

Comment (pyokagan) [reply src/test/java/seedu/address/model/task/DeadlineTaskTest.java:51]: Ugh I haven't taken a close look at this PR yet, sorry. (Too busy working on the new parser, which must must must be done by this cycle)

If you need this in ASAP, maybe get another reviewer?

Comment (nightism) [reply src/test/java/seedu/address/model/task/DeadlineTaskTest.java:51]: OK it's fine. This pr is not that emergent. Basically, what is does is quite similar to my previous pr which "marks the deadline task as finished".

Comment (pyokagan) [reply src/test/java/seedu/address/model/task/DeadlineTaskTest.java:51]: Yeah, this is definitely an improvement.

I think the testing for `equals()` that we do needs to be more comprehensive though (not needed for this PR). Then again, testing for all possibilities of all the fields is a lot of work. I wonder if we could use something like [EqualsVerifier](http://jqno.nl/equalsverifier/) for this.

File: src/main/java/seedu/address/logic/commands/MarkFloatingTaskFinishedCommand.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+                                                true);
+
+        try {
+            model.setFloatingTask(targetIndex - 1, finishedFloatingTask);
+        } catch (IllegalValueException e) {
+            throw new AssertionError("The target floating task cannot be missing", e);
+        }
+
+        return new CommandResult(String.format(MESSAGE_MARK_TASK_FINISHED_SUCCESS, finishedFloatingTask));
+
```
Comment (pyokagan) [inline src/main/java/seedu/address/logic/commands/MarkFloatingTaskFinishedCommand.java:46]: @nightism Can write a test for `MarkFloatingTaskFinishedCommand`? We really need to start to increase our test coverage, Do it in a separate PR.

**Review by pyokagan — APPROVED** [review]
Thanks, looks straightforward.

Take note that I expect to see a `MarkFloatingTaskFinishedCommand` in the future, along with #124

**Comment by pyokagan** [comment]
@nightism Yeah, it is extremely annoying I can't verify if this PR is actually correct or not because there are no tests ><

Will still merge though because I can't spot any errors for now.

**Comment by pyokagan** [comment]
@nightism By the way, we should have an `archive` command or something to delete all of the finished tasks.

Also, there is no way to `un-finish` a task.

**Comment by nightism** [comment]
Noted. @pyokagan Thanks for your comments.
